### PR TITLE
 [MIG] report_qweb_signer: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
This module extends the functionality of report module to sign
PDFs using a PKCS#12 certificate.

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check tests failure and runbot warning:
```
2018-11-13 09:49:28,314 152 WARNING openerp_test odoo.models: The model report.certificate has no _description
```

**Comment by chienandalu** [comment]
I have the tests failing on java's memory: https://travis-ci.org/OCA/reporting-engine/jobs/454946302#L783 Any travis configurations needed?

**Comment by pedrobaeza** [comment]
I'm trying one possible patch.

**Comment by pedrobaeza** [comment]
No luck. @Yajo do you guess something?

File: report_qweb_signer/data/defaults.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.config_parameter" id="report_qweb_signer_java_param">
+        <field name="key">report_qweb_signer.java_parameters</field>
+        <field name="value">-Xms4M -Xmx4M</field>
```
Comment (yajo) [inline report_qweb_signer/data/defaults.xml:5]: Try removing this default.

See https://github.com/OCA/reporting-engine/pull/253/commits/c82a549b72edcae277171f2db3518ad4183dfcd8

**Comment by chienandalu** [comment]
@Yajo Still the same: https://travis-ci.org/OCA/reporting-engine/jobs/456961880#L783

**Comment by yajo** [comment]
You can try adding this to `.travis.yml`, as explained in https://github.com/travis-ci/travis-ci/issues/8408:

```yaml
before_script:
  - _JAVA_OPTIONS=
```

**Comment by chienandalu** [comment]
Thanks @Yajo let's try that.

**Comment by chienandalu** [comment]
Nay :disappointed:

**Comment by yajo** [comment]
Could you try ideas from https://stackoverflow.com/a/33794801/1468388?

You could try running `ulimit -v` or similar inside Travis to know its limits...

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Merging as it's finally green

**Comment by etobella** [comment]
Just a question, I found a python library that could sign documents:
https://github.com/m32/endesive
Maybe we could check if we can replace the java. What do you think?

**Comment by pedrobaeza** [comment]
Well, this is working fine since a lot, so although Java is not my favorite one, unless this is proven to be work painless, I won't switch. We can try on the CS.

**Comment by etobella** [comment]
Ok, we will check it there :wink:

**Comment by chienandalu** [comment]
Hi, @etobella Yes, I discarded to try it because the project seemed to be a little immature yet. There was no much feedback or documentation around so adopting it seemed a little bit risky.

**Pull request merged**